mep3_localization/mep3_localization/robot_detection.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Memristor Robotics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cv2
+from cv2 import aruco
+from cv_bridge import CvBridge
+from geometry_msgs.msg import TransformStamped
+import numpy as np
+import rclpy
+from rclpy.node import Node
+from scipy.spatial.transform import Rotation as R
+from sensor_msgs.msg import CameraInfo, Image
+from tf2_ros import TransformBroadcaster
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+
+class DetectedRobots(Node):
+    """
+    Detects robots from the central tracking device.
+
+    This class receives video and camera parameters from the central RasPi Cam,
+    receives the static transform from map to camera in order to detect wrong
+    orientations and sends TF of Aruco tags.
+    """
+
+    def __init__(self):
+        super().__init__('image_subscriber')
+        self.image_subscription = self.create_subscription(
+            Image, '/cam/cam_central/RasPi0', self.image_listener_callback, 10)
+        self.image_subscription
+        self.camera_info_subscription = self.create_subscription(
+            CameraInfo, '/cam/cam_central/RasPi0/camera_info',
+            self.camera_info_listener_callback, 10)
+        self.camera_info_subscription
+        self._tf_publisher = TransformBroadcaster(self)
+        self.static_tf_listener('map', 'camera_static')
+
+        self.dict = aruco.Dictionary_get(aruco.DICT_4X4_100)
+        self.params = aruco.DetectorParameters_create()
+        # Side length of the ArUco marker in meters
+        self.aruco_robot_length = 0.07
+        self.aruco_42_length = 0.1
+
+    def image_listener_callback(self, data):
+        # self.get_logger().info('Receiving video frame')
+        br = CvBridge()
+        current_frame = br.imgmsg_to_cv2(data, 'bgr8')
+        rvecs, tvecs, ids = self.get_aruco_pose(current_frame)
+        self.send_pose_tf2(rvecs, tvecs, ids)
+
+    def static_tf_listener(self, parent, child):
+        # self.camera_translation = [-0.141, 1.417, 1.184]
+        # self.camera_rotation = R.from_quat([-0.005, 0.962, -0.272, 0.008])
+        self.tf_buffer = Buffer()
```

Comment:
Sve promjenjive treba inicijalizovati u konstruktoru, u najgorem slucaju inicijalizovati na `None` vrijednost

Reply:
> Jako dobro! Ima nekoliko manjih sitnica, ali me najvise brine kako cemo ovaj cvor iskoristiti i za detekciju markera sa kamere na robotu. Ja bih razbio na dva cvora, jedan koji samo obajvljuje transformacije i drugi koji koristi te transformacije za lokalizaciju robota

Zamislio sam da imamo tri čvora (klase):
1. `ArucoDetector` - učitava sliku i vraća nazad tf2 transformacije od Aruco markera koje pronađe. 
2. `PoseCorrector` - u ovom čvoru bi se rešavali problemi sa rotacijom Aruco markera, filtrirali bi se samo korektno orijentisani markeri. Pošto unapred znamo pozicije četiri Aruco markera na terenu (tagovi 20, 21, 22 i 23), mogli bismo te informacije da iskoristimo kako bismo bolje odredili pozicije ostalih markera. Kako to izvesti, da li neko ima ideju?
3. `ObjectLocalizer` - u ovom čvoru bi se implementirao Kalman filtar.